### ISSUE TITLE
`paren-spacing` should not result in compilation error by incorrect handling of EOL-comment

### ISSUE THREAD

**Issue by starsep** [pinterest/ktlint#3227]
## Expected Behavior
`ktlint` doesn't break Kotlin syntax.

## Observed Behavior
`ktlint` break syntax by moving `)` into part of comment

## Steps to Reproduce
1. Example.kt:
``` kotlin
fun main() {
  1.plus(1 // test
    ).plus(2 // test2
    ).plus(42)
}
```
2. Run `ktlint --format Example.kt`
3. Invalid Kotlin code:
``` kotlin
fun main() {
    1.plus(1 // test).plus(2 // test2).plus(42)
}
```

## Your Environment

* Version of ktlint used: 1.8.0
* Relevant parts of the `.editorconfig` settings: none
* Name and version (or code for custom task) of integration used (Gradle plugin, Maven plugin, command line, custom Gradle task): command line
* Version of Gradle used (if applicable): N/A
* Operating System and version: NixOS 25.11

**Comment by paul-dingemans** [comment]
Tnx for reporting. This is not a regression bug as the problem also exists in 1.7 release. Will fix it in next release.

**Comment by SubhrajyotiSen** [comment]
@paul-dingemans I would like to give this a shot

**Comment by paul-dingemans** [comment]
> @paul-dingemans I would like to give this a shot

Sure, go ahead.
